awsconfig/profiles.go

```diff
@@ -33,51 +56,83 @@ func loadAwsProfiles() {
 	}
 
 	awsProfiles = nil
+
+	// First load profiles from credentials file
 	if credentialsIniFile != nil {
 		for _, section := range credentialsIniFile.Sections() {
 			if section.Name() == ini.DefaultSection {
-				// AWS does not specify anything useful in the root section
 				continue
 			}
-			profile := Profile{
-				Name: section.Name(),
-			}
+
+			profileName := section.Name()
+			var region string
 
 			if configIniFile != nil {
-				var sectionName string
-				if profile.Name == "default" {
-					sectionName = profile.Name
+				var configSectionName string
+				if profileName == "default" {
+					configSectionName = profileName
 				} else {
-					// all other non-"default" profiles have special prefix
-					sectionName = "profile " + profile.Name
-				}
-				configProfileSection, _ := configIniFile.GetSection(sectionName)
-				if configProfileSection != nil {
-					regionKey, _ := configProfileSection.GetKey("region")
-					if regionKey != nil {
-						profile.Region = regionKey.Value()
-					}
+					configSectionName = "profile " + profileName
 				}
+				configSection, _ := configIniFile.GetSection(configSectionName)
+				region = getRegionFromSection(configSection)
+			}
+
+			addProfile(profileName, region)
+		}
+	}
+
+	// Then load SSO profiles from config file
+	if configIniFile != nil {
+		for _, section := range configIniFile.Sections() {
+			if section.Name() == ini.DefaultSection || !strings.HasPrefix(section.Name(), "profile ") {
+				continue
+			}
+
+			profileName := strings.TrimPrefix(section.Name(), "profile ")
+			if profileExists(profileName) {
+				continue
+			}
+
+			ssoSessionKey, _ := section.GetKey("sso_session")
+			ssoStartUrlKey, _ := section.GetKey("sso_start_url")
```

Comment:
can you add a test that covers these SSO paths? i know basically nothing about how these are configured, so it would be great to see a snapshot / example AWS config of what the workflow looks like when using SSO profiles + prevent regressions going forward. 

if this is difficult to add due to needing to actually interact with SSO, please explain how one would actually test this manually. i'm wondering if we can utilize test fixtures in somewhat to bypass having to interact with AWS in tests

Reply:
I have added config loading tests and added scenarios in the workflow tests for loading with SSO profile, SSO profile and overriding region etc. I hope this is what you meant, I'll update if not.